adapters/rubicon/rubicon.go

```diff
@@ -473,3 +503,174 @@ func NewRubiconAdapter(config *adapters.HTTPAdapterConfig, uri string, xuser str
 		XAPIPassword: xpass,
 	}
 }
+
+func (a *RubiconAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	numRequests := len(request.Imp)
+	errs := make([]error, 0, len(request.Imp))
+	var err error
+
+	requestData := make([]*adapters.RequestData, 0, numRequests)
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("User-Agent", "prebid-server/1.0")
+
+	for i := 0; i < numRequests; i++ {
+		request.Imp = request.Imp[i : i+1]
+
+		var bidderExt adapters.ExtImpBidder
+		if err = json.Unmarshal(request.Imp[0].Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		var rubiconExt openrtb_ext.ExtImpRubicon
+		if err = json.Unmarshal(bidderExt.Bidder, &rubiconExt); err != nil {
+			errs = append(errs, err)
+			return nil, errs
```

Comment:
if you want, all these error returns can be simplified to one line: `return nil, append(errs, err)`.

But also... do you _really_ mean to `return` rather than `continue` here? It seems to me like you may still want to bid on the other `Imp` objects, even if something goes wrong on one of them. If your adapter returns both `RequestData` _and_ `errors`, the core auction logic will do the best it can with them.

Reply:
you're absolutely right. continuing would make more sense here since we don't want to discard the other imps.